diff -urN a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2014-05-27 16:23:58.677624645 +0200
+++ b/src/CMakeLists.txt	2014-05-27 16:33:24.092639787 +0200
@@ -110,7 +110,7 @@
     ${ALVAR_HEADERS}
     ${ALVAR_SOURCES}
 )
-set_target_properties(ALVAR PROPERTIES OUTPUT_NAME alvar${ALVAR_VERSION_NODOTS})
+set_target_properties(ALVAR PROPERTIES OUTPUT_NAME alvar)
 set_target_properties(ALVAR PROPERTIES DEBUG_POSTFIX d)
 
 # link alvar library target to required libraries
--- a/src/CMakeLists.txt	2014-05-08 13:30:33.951229932 +0200
+++ b/src/CMakeLists.txt	2014-05-08 16:35:40.486772483 +0200
@@ -123,9 +123,10 @@
 if(ALVAR_PACKAGE MATCHES sdk)
     install(FILES ${ALVAR_HEADERS} ${ALVAR_GLOBAL_HEADER} DESTINATION include)
 endif(ALVAR_PACKAGE MATCHES sdk)
+install(FILES ${ALVAR_HEADERS} ${ALVAR_GLOBAL_HEADER} DESTINATION include/alvar)
 
 # install target
-install(TARGETS ALVAR DESTINATION bin)
+install(TARGETS ALVAR DESTINATION lib64)
 
 # set include directories and libraries variables for alvar
 list(APPEND ALVAR_INCLUDE_DIRS
