diff -ru gzweb.orig/get_local_models.py gzweb/get_local_models.py
--- gzweb.orig/get_local_models.py	2014-08-27 09:45:11.256045876 +0200
+++ gzweb/get_local_models.py	2014-08-27 09:47:11.058049085 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 from __future__ import print_function
 
 import sys
diff -ru gzweb.orig/webify_models.py gzweb/webify_models.py
--- gzweb.orig/webify_models.py	2014-08-27 09:45:11.257045876 +0200
+++ gzweb/webify_models.py	2014-08-27 09:45:39.040046620 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # To be deprecated and replaced by webify_models_v2.py
 
diff -ru gzweb.orig/webify_models_v2.py gzweb/webify_models_v2.py
--- gzweb.orig/webify_models_v2.py	2014-08-27 09:45:11.257045876 +0200
+++ gzweb/webify_models_v2.py	2014-08-27 09:45:33.667046477 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # A script to update gazebo models to be web-friendly.
 # It converts all textures to png format and make sure they are
